ledger/acctonline_test.go

```diff
@@ -1074,3 +1095,83 @@ func TestAcctOnlineCacheDBSync(t *testing.T) {
+	const compactCertVotersLookback = 2
+	const compactCertSecKQ = compactCertRounds / 2
+
+	const numAccts = maxBalLookback * 5
+	genesisAccts := []map[basics.Address]basics.AccountData{{}}
+	genesisAccts[0] = make(map[basics.Address]basics.AccountData, numAccts)
+	var addrA basics.Address
+	for i := 0; i < numAccts; i++ {
+		addr := ledgertesting.RandomAddress()
+		genesisAccts[0][addr] = ledgertesting.RandomOnlineAccountData(0)
+		if addrA.IsZero() {
+			addrA = addr
+		}
+	}
+
+	pooldata := basics.AccountData{}
+	pooldata.MicroAlgos.Raw = 100 * 1000 * 1000 * 1000 * 1000
+	pooldata.Status = basics.NotParticipating
+	genesisAccts[0][testPoolAddr] = pooldata
+
+	sinkdata := basics.AccountData{}
+	sinkdata.MicroAlgos.Raw = 1000 * 1000 * 1000 * 1000
+	sinkdata.Status = basics.NotParticipating
+	genesisAccts[0][testSinkAddr] = sinkdata
+
+	testProtocolVersion := protocol.ConsensusVersion("test-protocol-TestAcctOnlineCacheDBSync")
+	protoParams := config.Consensus[protocol.ConsensusFuture]
+	protoParams.MaxBalLookback = maxBalLookback
+	protoParams.SeedLookback = seedLookback
+	protoParams.SeedRefreshInterval = seedInteval
+	protoParams.CompactCertRounds = compactCertRounds
+	protoParams.CompactCertVotersLookback = compactCertVotersLookback
+	protoParams.CompactCertSecKQ = compactCertSecKQ
+	config.Consensus[testProtocolVersion] = protoParams
+	defer func() {
+		delete(config.Consensus, testProtocolVersion)
+	}()
+
+	ml := makeMockLedgerForTracker(t, true, 1, testProtocolVersion, genesisAccts)
+	defer ml.Close()
+	conf := config.GetDefaultLocal()
+
+	au, oa := newAcctUpdates(t, ml, conf, ".")
+	defer oa.close()
+	_, totals, err := au.LatestTotals()
+	require.NoError(t, err)
+
+	// add maxBalLookback empty blocks
+	maxBlocks := maxBalLookback * 5
+	for i := 1; i <= maxBlocks; i++ {
+		var updates ledgercore.AccountDeltas
+		updates.Upsert(addrA, ledgercore.AccountData{AccountBaseData: ledgercore.AccountBaseData{Status: basics.Online}, VotingData: ledgercore.VotingData{VoteLastValid: basics.Round(100 * i)}})
+		base := genesisAccts[i-1]
+		newAccts := applyPartialDeltas(base, updates)
+		totals = newBlock(t, ml, totals, testProtocolVersion, protoParams, basics.Round(i), base, updates, totals)
+		genesisAccts = append(genesisAccts, newAccts)
+		commitSync(t, oa, ml, basics.Round(i))
+	}
+	require.Len(t, oa.deltas, int(conf.MaxAcctLookback))
+	require.Equal(t, basics.Round(maxBlocks-int(conf.MaxAcctLookback)), oa.cachedDBRoundOnline)
+	// voters stalls after the first interval
+	lowest := oa.voters.lowestRound(oa.cachedDBRoundOnline)
+	require.Equal(t, basics.Round(compactCertRounds-compactCertVotersLookback), lowest)
+	require.Equal(t, maxBlocks/compactCertRounds, len(oa.voters.round))
+	// lowestRound is fixed but online accounts tracker successfully decreases the history before the lowestRound
+	// TODO: add max bound check after implementing it for voters tracker
```

Comment:
when we have the max X value, we might enforce in online accounts tracker and check it here